File: SecondSemester/lab2/C#/Line.cs
using System;

namespace lab2 {
    public class Line {
        private char[] line;
        public Line(char[] line) {
            this.line = line;
        }

        public void outputString() {
            for (int i = 0;i < this.getLength();i++) {
                Console.Write(line[i]);
            }
        }

        public int getLength() {
            int i = 1;
            for (;i < line.Length;i++)
                if (line[i] == default(char))
                    break;
            
            return i;
        }

        public int findElement(char obj) {
            int counter = 0;

            for (int i = 0;i < this.getLength();i++) {
                if (line[i] == obj)
                    counter++;
            }

            return counter;
        }

        public bool compare(Line obj) {
            bool equal = true;
            for (int i = 0;i < this.getLength(); i++) {
                Console.WriteLine(this.line[i] + " = " + obj.line[i]);
                if (this.line[i] != obj.line[i])
                    equal = false;
            }

            return equal;
        }
    }
}

File: SecondSemester/lab2/C++/Line.cpp
#include <iostream>
#include <cstring>
#include "Line.h"

using namespace std;

Line::Line(char* Line) {
    this->line = Line;
}

int Line::getLength() {
    return strlen(line);
}

int Line::findElement(char obj) {
    int times = 0;

    for(int j = 0; j < strlen(line); j++) {
        if(line[j] == obj) {
            times += 1;
        }
    }

    return times;
}

void Line::outputString() {
    for(int j = 0; j < strlen(line); j++)
        cout << line[j];
    cout << endl;
}